### PR TITLE
Improve component prop possible values in <Route /> docs

### PR DESCRIPTION
**Pull request by aarongarciah**
Add an example of using a function as value of the component prop in <Route /> docs.

### CODE REVIEW

File: packages/react-router/docs/api/Route.md [hunk 1]
```diff
@@ -69,7 +69,17 @@ const User = ({ match }) => {
...
+When you use `component` (instead of `render` or `children`, below) the router uses [`React.createElement`](https://facebook.github.io/react/docs/react-api.html#createelement) to create a new [React element](https://facebook.github.io/react/docs/rendering-elements.html) from the given component.
+
+That means if you provide an inline function to the `component` prop, you would create a new component every render. This results in the existing component unmounting and the new component mounting instead of just updating the existing component. When using an inline function for inline rendering, use the `render` or the `children` prop (below).
+
+```js
+<Route path="/user/:username" component={() => <User />}/>
```
Comment (kutyel) [inline packages/react-router/docs/api/Route.md]: I think you receive the match from exactly that arrow function:
```js
<Route path="/user/:username" component={({ match }) => <User match={match} />} />
```

Comment (aarongarciah) [inline packages/react-router/docs/api/Route.md]: You're right, better to add it to the example.

File: packages/react-router/docs/api/Route.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ All three [render methods](#route-render-methods) will be passed the same three
 - [location](./location.md)
 - [history](./history.md)
 
-## component
+## component: component | func
```
Comment (timdorr) [inline packages/react-router/docs/api/Route.md]: A function is a component and a component is a function. This isn't needed.

File: packages/react-router/docs/api/Route.md [hunk 3]
```diff
@@ -69,7 +69,17 @@ const User = ({ match }) => {
...
+When you use `component` (instead of `render` or `children`, below) the router uses [`React.createElement`](https://facebook.github.io/react/docs/react-api.html#createelement) to create a new [React element](https://facebook.github.io/react/docs/rendering-elements.html) from the given component.
+
+That means if you provide an inline function to the `component` prop, you would create a new component every render. This results in the existing component unmounting and the new component mounting instead of just updating the existing component. When using an inline function for inline rendering, use the `render` or the `children` prop (below).
+
+```js
+<Route path="/user/:username" component={(props) => <User {...props} />}/>
```
Comment (timdorr) [inline packages/react-router/docs/api/Route.md]: This isn't using the `render` or `children` prop like the docs say above.

File: packages/react-router/docs/api/Route.md (hunk 2 above)
Comment (aarongarciah) [inline packages/react-router/docs/api/Route.md]: You're right, but in the render prop is specified `func`, so I thought it was a good idea to add it.

File: packages/react-router/docs/api/Route.md (hunk 3 above)
Comment (aarongarciah) [inline packages/react-router/docs/api/Route.md]: The point is you can can use this to force the unmounting/mounting of this component in every render. I understand this is rarely desired, but I'm in a project where I've done this to accomplish what I needed. I'll provide an example later.

Comment (pshrmn) [inline packages/react-router/docs/api/Route.md]: I don't think that the documentation should show `<Route component={() => {...}} />` anywhere. If you need to force React to create a new component, adding a `key` prop is a much better approach.

**Comment by aarongarciah** [comment]
I see that this PR isn't necessary. Thanks for your time!

**Pull request closed without merging**